python/cudf/cudf/_lib/copying.pyx

```diff
@@ -204,6 +204,9 @@ def _scatter_scalar(scalars, Column scatter_map,
     cdef Scalar slr
     for val, col in zip(scalars, target_table._columns):
         slr = as_scalar(val, col.dtype)
+        # Since this uses this want the unique_ptr but not
+        # the value, we need an explicit sync
+        slr._sync()
```

Comment:
From my perspective we shouldn't ever need to manually sync the scalar. We should have accessor methods for the host value and device value and those methods should be responsible for syncing the scalar if necessary.

Reply:
Should we be using `get_c_value()` instead of `.c_value`?

Reply:
This is a strange case because `get_c_value()` is meant to return a `scalar *`, which is sufficient in most cases. But here since it's trying to construct the argument to scatter which has to be a `vector[unique_ptr[scalar]]` which is what we would get from `c_value`. 

The problem is that it's not safe to access `c_value` anymore because of the lazy execution. If it isn't set it can segfault here. `get_c_value` was hopefully supposed to be the safe version that syncs first. 

I think the best solution would be another "safe function" that yields the unique pointer.